packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -171,7 +215,61 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
     parentRef.current.focus();
   };
 
+  setSelectedItemIds = (e: React.SyntheticEvent, selectedItemIds: string[]) => {
+    _.invoke(this.props, 'onSelectedItemIdsChange', e, { ...this.props, selectedItemIds });
+
+    this.setState({
+      selectedItemIds,
+    });
+  };
+
+  processItemsForSelection = (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
+    let { selectedItemIds } = this.state;
+    const { id, selectableParent, items, expanded } = treeItemProps;
+    const treeItemHasSubtree = hasSubtree(treeItemProps);
+    const isExpandedSelectableParent = treeItemHasSubtree && selectableParent && expanded;
+
+    // if the parent is not selectable or is collapsed it means treeItem should be expanded, not procced with selection
+    if ((treeItemHasSubtree && !selectableParent) || (treeItemHasSubtree && !expanded)) {
```

Comment:
`treeItemHasSubtree && (!selectableParent || !expanded)` or even better `treeItemHasSubtree && !(selectableParent && expanded)` . Would that work? :)

Reply:
I'd go with the first suggestion, for me it's more readable